### PR TITLE
Implement Webauthn registration

### PR DESCRIPTION
**Pull request by codingllama**
Add server-side Webauthn registration logic to lib/auth/webauthn.

This allows both U2F and Webauthn-enabled devices to be registered.

### CODE REVIEW

**Comment by codingllama** [comment]
Branched from https://github.com/gravitational/teleport/pull/8201 - once I can get it merged I'll rebase and change the merge target to master.

File: lib/auth/webauthn/login_test.go [hunk 1]
```diff
@@ -33,95 +33,127 @@ import (
...
-	require.Len(t, identity.UpdatedDevices, 1)
-	got := identity.UpdatedDevices[0]
-	if diff := cmp.Diff(wantDev, got); diff != "" {
-		t.Errorf("Updated device mismatch (-want +got):\n%s", diff)
+	tests := []struct {
+		name         string
+		user, origin string
+		key          *mocku2f.Key
+	}{
+		{
```
Comment (zmb3) [inline lib/auth/webauthn/login_test.go:83]: These two tests look identical to me.

File: lib/auth/webauthn/register.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+	sessionDataPB, err := sessionToPB(sessionData)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := f.Identity.UpsertWebauthnSessionData(ctx, user, registrationSessionID, sessionDataPB); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return (*CredentialCreation)(credentialCreation), nil
```
Comment (zmb3) [inline lib/auth/webauthn/register.go:123]: Why is this cast necessary?

File: lib/auth/webauthn/login_test.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/webauthn/login_test.go:83]: My bad, fixed.

File: lib/auth/webauthn/register.go (hunk 2 above)
Comment (codingllama) [reply lib/auth/webauthn/register.go:123]: Casting from (duo-labs)wan.CredentialCreation to our (lib.auth.)webauthn.CredentialCreation. The latter is a type based on the former.

Comment (zmb3) [reply lib/auth/webauthn/register.go:123]: Got it, thanks!

**Review by zmb3 — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks, everyone.

**Pull request merged**